lib/shopify_app/session/session_storage.rb

```diff
@@ -16,5 +16,17 @@ def with_shopify_session(&block)
         &block
       )
     end
+
+    def access_scopes=(scopes)
+      super(scopes)
+    rescue NotImplementedError, NoMethodError
+      Rails.logger.warn("#access_scopes= must be defined to handle storing access scopes: #{scopes}")
```

Comment:
I don't think this is a great pattern. Now a) the rest of the system needs to support both the "can store access scopes" and "can not store access scopes" states indefinitely and b) the developer might think they properly implemented access scope support and the only sign that they messed it up is `warn` error messages.

It would be better IMO to create a `ShopSessionStorageWithScopes` interface that adds the `access_scopes` support (and errors if it doesn't work) and then to change behaviour elsewhere based on which interface is active. Then we make a breaking change in a future major version where we say we're only supporting the new interface.

Reply:
This makes sense. I'll refactor this piece. Thanks again Mike!